src/lib.rs

```diff
@@ -45,6 +45,92 @@ pub fn parse_file(path: &PathBuf) -> Result<(), pest::error::Error<Rule>> {
     }
 }
 
+pub fn parse_graph_stages(path: &PathBuf) -> Result<(), PestError<Rule>> {
+    use std::fs::File;
+    use std::io::Read;
+
+    match File::open(path) {
+        Ok(mut file) => {
+            let mut contents = String::new();
+
+            if let Err(e) = file.read_to_string(&mut contents) {
+                return Err(PestError::new_from_pos(
+                    ErrorVariant::CustomError {
+                        message: format!("{}", e),
+                    },
+                    pest::Position::from_start(""),
+                ));
+            } else {
+                println!("{}", build_dot_file(stage_graphs(&contents)?));
+                Ok(())
+            }
+        }
+        Err(e) => {
+            return Err(PestError::new_from_pos(
+                ErrorVariant::CustomError {
+                    message: format!("{}", e),
+                },
+                pest::Position::from_start(""),
+            ));
+        }
+    }
+}
+
+fn build_dot_file<'a>(stages: Vec<&str>) -> String {
+    let nodes = stages.windows(2).fold("".to_string(), |acc, stage| {
+        let stage1 = stage[0];
+        let stage2 = stage[1];
+
+        if acc == "".to_string() {
+            return format!("\"{}\" -> \"{}\";", stage1, stage2);
+        }
+        format!("{}\"{}\" -> \"{}\";", acc, stage1, stage2)
+    });
+    format!(r#"digraph {{ {} }}"#, nodes)
+}
+
+fn stage_graphs(buffer: &str) -> Result<Vec<&str>, PestError<Rule>> {
```

Comment:
@himanoa how does this work with a set of parallel stages?  I can easily imagine what the dot syntax looks like for those ,but I'm not quite understanding how that might look with a flat Vec here